pilot/pkg/serviceregistry/kube/controller/ambientindex_test.go

```diff
@@ -818,58 +925,73 @@ func addPolicy(t *testing.T, cfg *memory.Controller, name, ns string, selector m
-	t.Helper()
-	want := sets.New(names...)
-	assert.EventuallyEqual(t, func() sets.String {
-		var addresses []*model.AddressInfo
-		if lookup == "" {
-			addresses = controller.ambientIndex.All()
-		} else {
-			addresses = controller.ambientIndex.Lookup(lookup)
-		}
-		have := sets.New[string]()
-		for _, address := range addresses {
-			switch addr := address.Address.Type.(type) {
-			case *workloadapi.Address_Workload:
-				have.Insert(addr.Workload.Name)
-			case *workloadapi.Address_Service:
-				have.Insert(addr.Service.Name)
-			}
-		}
-		return have
-	}, want, retry.Timeout(time.Second*3))
-}
-
-func deletePod(t *testing.T, pc clienttest.TestClient[*corev1.Pod], name string) {
+func (s *ambientTestServer) deletePod(t *testing.T, name string) {
 	t.Helper()
-	pc.Delete(name, "ns1")
+	s.pc.Delete(name, testNS)
 }
 
-func assertEvent(t *testing.T, fx *xdsfake.Updater, ip ...string) {
+func (s *ambientTestServer) assertEvent(t *testing.T, ip ...string) {
 	t.Helper()
 	want := strings.Join(ip, ",")
-	fx.MatchOrFail(t, xdsfake.Event{Type: "xds", ID: want})
+	s.fx.MatchOrFail(t, xdsfake.Event{Type: "xds", ID: want})
 }
 
-func deleteService(t *testing.T, sc clienttest.TestClient[*corev1.Service], name string) {
+func (s *ambientTestServer) deleteService(t *testing.T, name string) {
 	t.Helper()
-	sc.Delete(name, "ns1")
+	s.sc.Delete(name, testNS)
 }
 
-func addService(t *testing.T, sc clienttest.TestClient[*corev1.Service], name string, labels, annotations map[string]string,
+func (s *ambientTestServer) addService(t *testing.T, name string, labels, annotations map[string]string,
 	ports []int32, selector map[string]string, ip string,
 ) {
 	t.Helper()
-	service := generateService(name, "ns1", labels, annotations, ports, selector, ip)
-	sc.CreateOrUpdate(service)
+	service := generateService(name, testNS, labels, annotations, ports, selector, ip)
+	s.sc.CreateOrUpdate(service)
+}
+
+func (s *ambientTestServer) lookup(key string) []*model.AddressInfo {
+	if key == "" {
+		return s.controller.ambientIndex.All()
+	}
+	return s.controller.ambientIndex.Lookup(key)
+}
+
+func (s *ambientTestServer) clearEvents() {
+	s.fx.Clear()
+}
+
+// Returns the resource name for the given pod.
+func (s *ambientTestServer) pod(name string) string {
+	return fmt.Sprintf("%s//Pod/%s/%s",
+		s.controller.clusterID, testNS, name)
+}
+
+// Returns the resource name for the given address.
+func (s *ambientTestServer) addr(addr string) string {
+	return string(s.controller.network) + "/" + addr
+}
+
+// Returns the resource name for the given service.
+func (s *ambientTestServer) svc(name string) string {
+	return fmt.Sprintf("%s/%s", testNS, s.hostname(name))
+}
+
+// Returns the resource name for the given service.
+func (s *ambientTestServer) hostname(serviceName string) string {
+	return fmt.Sprintf("%s.%s.svc.company.com", serviceName, testNS)
+}
+
+// / Returns the resource name for the given WorkloadEntry.
+func (s *ambientTestServer) wle(wleName string) string {
```

Comment:
```suggestion
func (s *ambientTestServer) wleEvent(wleName string) string {
```

Reply:
I'm on the fence about this change. I'm not sure "event" is sufficiently more descriptive to warrant the extra verbosity at the call sites. I'm open though if folks feel strongly here.

Reply:
`wleXdsName`?